app/src/analytics/analytics-client.ts

```diff
@@ -1,91 +1,168 @@
 import crypto from 'crypto';
+import fetch from 'node-fetch';
+import { v4 as uuidv4 } from 'uuid';
 
 interface AnalyticsAttributes {
-    [key: string]: any;
+	[key: string]: any;
 }
 
-interface EventPayload {
-    eventName: string,
-    action: string,
-    actionSubject: string,
-    attributes?: AnalyticsAttributes
+interface Metadata {
+	nodeVersion: string;
+	version: string;
+}
+
+interface Context {
+	library: any
 }
 
-enum EnvType {
-    LOCAL = 'local',
-    DEV = 'dev',
-    STAGING = 'staging',
-    PROD = 'prod'
+interface EventPayload {
+	action: string,
+	actionSubject: string,
+	attributes?: AnalyticsAttributes
 }
 
+interface Event {
+	userId?: string;
+	anonymousId: string;
+	event: string;
+	properties: {
+		action: string;
+		actionSubject: string;
+		attributes: Record<string, string>;
+		env: string;
+		eventType: string;
+		origin: string;
+		product: string;
+		tenantId: string;
+		tenantIdType: string;
+		source: string;
+		userIdType?: string;
+	};
+	type: string;
+	timestamp: string;
+	_metadata: Metadata;
+	context: Context;
+	messageId: string;
+  }
+
+interface Payload {
+	batch: Event[];
+	timestamp: string;
+	sentAt: string;
+  }
+
+interface Options {
+	method: string;
+	body: string;
+	headers: {
+		'Content-Type': string;
+	};
+  }
+
+  const environments = {
+	LOCAL: 'local',
+	DEV: 'dev',
+	STAGING: 'stg',
+	PROD: 'prod'
+  };
+
+  const TRACK_EVENT_TYPE = 'track';
+  const PRODUCT = 'jenkinsForJira';
+  const ORIGIN = 'server';
+  const TENANT_ID_TYPE = 'cloudId';
+
 // eslint-disable-next-line max-len
 export const sendAnalytics = async (cloudId: string, eventPayload: EventPayload, accountId?: string, anonymousId?: string): Promise<void> => {
-    sendEvent(cloudId, eventPayload, accountId, anonymousId)
-        .then(() => console.log('Analytics event processed'))
-        .catch((e) => console.error({ e }, 'Failed to send analytics event'));
+	console.info('Analytics Request');
+
+	sendEvent(cloudId, eventPayload, accountId, anonymousId)
+		.then(() => console.info('Analytics event processed'))
+		.catch((e) => console.error({ e }, 'Failed to send analytics event'));
 };
 
-// eslint-disable-next-line max-len
+const getAnalyticsEnvironmentUrl = () => {
+	if (isProductionEnv()) {
+        return process.env.ANALYTICS_URL;
```

Comment:
/nit indentation

Reply:
As this is quite a small function you could rewrite it to be a var
```
const analyticsEnvrionmentUrl = isProductionEnv() ? process.env.ANALYTICS_URL : process.env.ANALYTICS_STAGE_URL;
```

Reply:
indent appears to be GitHub lying, codes sweet